static/js/util/util.js

```diff
@@ -168,30 +169,58 @@ export function makeCourseProgressDisplay(course, isFirst, isLast) {
  * @param {Object} profile The user profile
  * @returns {Object} Validation errors or an empty object if no errors
  */
-export function validateProfile(profile) {
+export function validateProfile(profile, requiredFields, messages) {
+  let clone = _.cloneDeep(profile);
+  let fields = Object.entries(clone).filter(pair => _.isObject(pair[1]));
+  let nullObj = (obj) => {
+    for (let key in obj) { obj[key] = null };
+  }
+  fields.map( (pair) => {
+    for (let obj of pair[1]) {
+      nullObj(obj);
+    }
+  });
   let errors = {};
+  fields.map( pair => errors[pair[0]] = pair[1]);
```

Comment:
Can you use `forEach` instead of `map`? There's not much difference but `map` usually indicates you wanted to have no side effects and return the changed value, while `forEach` is explicitly for causing side effects of some kind (like populating the `errors` object above)